nima_core/embeddings.py

```
@@ -110,44 +108,37 @@ def embed_batch(texts: list[str]) -> list[list[float]]:
-            f"[nima_core.embeddings] Unexpected Ollama response (no 'embeddings' key): {data}"
+    results_list: list[list[float]] = []
+    for txt in safe_texts:
+        payload = json.dumps(
+            {"model": EMBEDDING_MODEL, "prompt": txt}
+        ).encode("utf-8")
+        req = urllib.request.Request(
+            _EMBED_URL,
+            data=payload,
+            headers={"Content-Type": "application/json"},
+            method="POST",
         )
```

Ruff (0.15.10)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)